### PR TITLE
Document theory in hpmc

### PR DESCRIPTION
**Pull request by joaander**
## Description

This is a full edit pass on the documentation of the `hoomd.hpmc` package, with special attention to ensuring that the actions of operations are formally explained with mathematical notation (https://github.com/glotzerlab/hoomd-blue/issues/1078).

## Motivation and context

Ensure that the documentation is clear, consistent, accurate, and unambiguous.

## How has this been tested?

I inspected the rendered documentation locally.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
Note: 
* #1017 documents the HPMC tuners.
* `hoomd.hpmc.update.Clusters` will be documented later #1253.
* `hoomd.hpmc.update.MuVT` will be documented later #1253.

File: hoomd/hpmc/external/user.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
 # Copyright (c) 2009-2022 The Regents of the University of Michigan.
 # Part of HOOMD-blue, released under the BSD 3-Clause License.
 
-"""User-defined external fields for HPMC simulations."""
+"""User-defined external fields for HPMC simulations.
```
Comment (b-butler) [inline hoomd/hpmc/external/user.py:4]: I can't remember, but isn't this only supported on CPU? If so we should document it here.

File: hoomd/hpmc/external/wall.py [hunk 2]
```diff
@@ -95,8 +95,8 @@ class WallPotential(ExternalField):
 
     .. math::
 
-        U_{\mathrm{walls}} = \sum_{i=0}^{N_{\mathrm{particles}}}
-        \sum_{j=0}^{N_{\mathrm{walls}}} U_{i,j},
+        U_{\mathrm{walls}} = \sum_{i=0}^{N_{\mathrm{particles}-1}}
```
Comment (b-butler) [inline hoomd/hpmc/external/wall.py:98]: We only support CPU execution here as well correct?

**Review by b-butler — CHANGES_REQUESTED** [review]
I only have small comments. The documentation changes are top notch!

File: hoomd/hpmc/update.py [hunk 3]
```diff
@@ -25,16 +32,185 @@ class BoxMC(Updater):
...
+      :math:`[-\delta_\mathrm{volume}, \delta_\mathrm{volume}]`.
+
+      In 2D:
+
+      .. math::
+
+          V^t &= V + u \\
+          L_x^t &= \left( \frac{Lx}{Ly} V^t \right)^{1/2} \\
+          L_y^t &= L_x^t \frac{Ly}{Lx} \\
+          xy^t &= xy \\
```
[the remark this answers is not in the document]
Comment (joaander) [reply hoomd/hpmc/update.py:108]: Indeed.

File: hoomd/hpmc/pair/user.py [hunk 4]
```diff
@@ -187,7 +202,7 @@ class CPPPotential(CPPPotentialBase):
 
     Args:
         r_cut (float): Particle center to center distance cutoff beyond which
-            all pair interactions are assumed 0.
+            all pair interactions are 0.
```
[the remark this answers is not in the document]
Comment (joaander) [reply hoomd/hpmc/pair/user.py:205]: Fixed.

File: hoomd/hpmc/external/user.py (hunk 1 above)
Comment (joaander) [reply hoomd/hpmc/external/user.py:4]: This is documented on line 82:
```
    Note:
        `CPPExternalPotential` does not support execution on GPUs.
```

File: hoomd/hpmc/external/wall.py (hunk 2 above)
Comment (joaander) [reply hoomd/hpmc/external/wall.py:98]: Correct. I added a note.

**Review by b-butler — APPROVED** [review]
:+1:

**Pull request merged**